vllm_spyre/v1/core/scheduler.py

```diff
@@ -109,6 +113,88 @@ def schedule(self) -> "SchedulerOutput":
         outputs = super().schedule()
         return outputs
 
+    def schedule_cb(self) -> "SchedulerOutput":
+        """This override adds constraints and then delegates most of the work
+        to the base scheduler"""
+
+        # First purge the full waiting queue into our holdback queue, preserving
+        # priority
+        while self.waiting:
+            self.holdback_queue.append(self.waiting.popleft())
+
+        # Check if there are new request and if total scheduled running request are less
+        # than the max number
+        total_running = self.last_running + self.running
+
+        if len(self.holdback_queue) > 0 and len(total_running) + 1 <= self.max_num_running_reqs:
+            # Make a copy of the warmup shapes
+            available_warmup_shapes = list(self.spyre_warmup_shapes)
+
+            while self.holdback_queue:
+                request = self.holdback_queue[0]
+
+                # prune the possible shapes to only those that fit this request
+                # and the growing batch size
+                max_tokens = 0
+                if request.sampling_params is not None and\
+                        request.sampling_params.max_tokens is not None:
+                    max_tokens = request.sampling_params.max_tokens
+
+                available_warmup_shapes = [
+                    shape for shape in available_warmup_shapes
+                    if request.num_prompt_tokens <= shape['prompt_length']
+                    and max_tokens <= shape['new_tokens']
+                    and len(self.waiting) < shape['batch_size']
+                ]
```

Comment:
I think that the continuous batching logic should not depend on the warmup shapes in this way?